org.mwc.cmap.NaturalEarth/src/org/mwc/cmap/naturalearth/wrapper/NELayer.java

```diff
@@ -0,0 +1,450 @@
+package org.mwc.cmap.naturalearth.wrapper;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.Graphics2D;
+import java.awt.Rectangle;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.core.runtime.IStatus;
+import org.geotools.data.FileDataStore;
+import org.geotools.data.FileDataStoreFinder;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.map.DirectLayer;
+import org.geotools.map.FeatureLayer;
+import org.geotools.map.MapContent;
+import org.geotools.map.MapViewport;
+import org.geotools.renderer.lite.RendererUtilities;
+import org.geotools.styling.FeatureTypeStyle;
+import org.geotools.styling.PolygonSymbolizer;
+import org.geotools.styling.Rule;
+import org.geotools.styling.SLD;
+import org.geotools.styling.Style;
+import org.geotools.styling.Symbolizer;
+import org.mwc.cmap.gt2plot.data.GeoToolsLayer;
+import org.mwc.cmap.naturalearth.Activator;
+import org.mwc.cmap.naturalearth.NaturalearthUtil;
+import org.mwc.cmap.naturalearth.view.NEFeatureRoot;
+
+import MWC.GUI.BaseLayer;
+import MWC.GUI.CanvasType;
+import MWC.GUI.Editable;
+import MWC.GUI.Layer;
+import MWC.GUI.Plottable;
+import MWC.GUI.Shapes.ChartBoundsWrapper;
+import MWC.GenericData.WorldArea;
+import MWC.GenericData.WorldLocation;
+
+public class NELayer extends GeoToolsLayer implements BaseLayer.ProvidesRange
+{
+
+	private static final long serialVersionUID = 1L;
+	
+	private static final int BATHY_HEIGHT = 20;
+
+	/** from StreamingRenderer - Tolerance used to compare doubles for equality */
+  private static final double TOLERANCE = 1e-6;
+
+
+	private NEFeatureRoot _myFeatures;
+	//private Layers _theLayers;
+
+	private List<FeatureLayer> _gtLayers = new ArrayList<FeatureLayer>();
+	
+	private Map<String, Color> _bathyKeys = new LinkedHashMap<String, Color>();
+
+	private double _maxScale = 0;
+	private double _minScale = Double.MAX_VALUE;
+
+	public NELayer(NEFeatureRoot features)
+	{
+		super(ChartBoundsWrapper.NELAYER_TYPE, NATURAL_EARTH, null);
+		setName(NATURAL_EARTH);
+		_myFeatures = features;
+	}
+	
+	@Override
+	public void clearMap()
+	{
+		for (FeatureLayer layer : _gtLayers)
+		{
+			//layer.dispose();
+			_myMap.removeLayer(layer);
+		}
+		
+		_gtLayers.clear();
+	}
+
+
+	@Override
+	public void setMap(MapContent map)
+	{
+		// store the map object.  
+		_myMap = map;
+		DirectLayer background = null;
+		// ok, now sort out which dataset we're looking at
```

Comment:
the following getVisible() causes troubles.  setMap() only gets called once, when NE is loaded.

If the user has NE in his plot, but saves the plot with NE visible, this block of logic doesn't get processed - and it's not possible to switch NE on.

So, IMHO we should perform this NE loading functionality - even if NE isn't visible.

Ok, fixed in 511cff75af017b856346daed492cdf9625e63cc5